x-pack/plugins/data_enhanced/server/search/session/types.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ShardsResponse } from 'elasticsearch';
+
+export enum SearchStatus {
+  IN_PROGRESS = 'in_progress',
+  ERROR = 'error',
+  COMPLETE = 'complete',
+}
+
+export interface AsyncSearchStatusResponse<T = unknown> {
```

Comment:
Should this extend `Omit<AsyncSearchResponse, 'response'>`? We should probably move `start_time_in_millis` and `expiration_time_in_millis` there anyway.